frontend/rolecall/src/app/piece/piece_editor.component.ts

```diff
@@ -88,22 +91,37 @@ export class PieceEditor implements OnInit {
   }
 
   private buildRenderingList() {
-    this.renderingItems = this.workingPieces.map(
-        (workingPiece, workingPieceIndex) => {
-      const hasNoChidren = workingPiece.type === 'SEGMENT'
-          ? false : workingPiece.positions.length === 0;
-      const sortPrefix = hasNoChidren ? ' ' : 'z';
-      const name = hasNoChidren ? '*' + workingPiece.name : workingPiece.name;
+    // Remove Super Ballet children
+    this.displayedPieces = this.workingPieces.filter(piece => !piece.siblingId);
+    this.displayedPieces.sort((a, b) => a.name < b.name ? -1 : 1);
```

Comment:
Hmm, it's too bad that we have to do all this work manually to figure out what is visible and what isn't. 

Is there some existing UI component we can reuse managing layout and visibility state? I'm not familiar with the component library we are using here, but I know the Angular Material components have [expansion panels](https://material.angular.io/components/expansion/overview) and [trees](https://material.angular.io/components/tree/overview), those would be the first things I would look for if I was to implement similar functionality (if something similar is available).

Reply:
The hiding and closing of the 'Super Ballet' children is the easy part. The tricky part is to show them in the right order etc. (For instance, the overall list is supported to be sorted in alphabetic order, while the children of the Super Ballet are supposed to be sorted in performance order.) Therefore I believe we would have to write most of this code again, even if we had an expansion panel. In our case, we are purposefully reusing the exact existing display object. The logic is not really to figure out how display and hide, the logic is figuring out which objects should be shown or hidden.

Reply:
We can revisit this one in the future to see if it's worth looking into.